### PR TITLE
[Enhancement] add appProtocol in fe service

### PR DESCRIPTION
**Pull request by yandongxiao**
Fixes: https://github.com/StarRocks/starrocks-kubernetes-operator/issues/287

### CODE REVIEW

File: pkg/common/resource_utils/service.go [hunk 1]
```diff
@@ -108,13 +125,20 @@ func BuildExternalService(src *srapi.StarRocksCluster, name string, serviceType
...
-		ports = append(ports, corev1.ServicePort{
+		servicePort := corev1.ServicePort{
 			Name:       sp.Name,
 			Port:       sp.Port,
 			NodePort:   sp.NodePort,
 			Protocol:   corev1.ProtocolTCP,
 			TargetPort: intstr.FromInt(int(sp.ContainerPort)),
-		})
+		}
+		if servicePort.Name == FeQueryPortName {
```
Comment (kevincai) [inline pkg/common/resource_utils/service.go:135]: how about the webPort? does it need to add appProtocol `http`?

Comment (kevincai) [reply pkg/common/resource_utils/service.go:135]: didn't see our webport related to the server-first-protocol document. HTTP protocol doesn't look like a server-first-protocol per the description in the doc.

Comment (yandongxiao) [reply pkg/common/resource_utils/service.go:135]: MySQL is a server-first-protocol, so we only need to add appProtocol to the query port, 9030.

**Review by kevincai — DISMISSED** [review]

**Comment by kevincai** [comment]
ci operator failure, might need take a look @yandongxiao

**Review by kevincai — APPROVED** [review]

**Pull request merged**